### PR TITLE
[docs] Improved development installation instructions order

### PR DESCRIPTION
**Pull request by praptisharma28**
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [ ] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [x] I have updated the documentation.

## Description of Changes

The changes make the installation process more straightforward by:
1. Showing the big picture (directory structure) before details
2. Ensuring proper package isolation through correct virtualenv setup
3. Following Python best practices for development environments

### CODE REVIEW

**Review by pandafy — CHANGES_REQUESTED** [review]
@praptisharma28 I would just make the instructions consistent with other modules. 

You can use the [openwisp-controller](https://openwisp.io/docs/dev/controller/developer/installation.html#installing-for-development) for reference. 

https://github.com/openwisp/openwisp-controller/blob/master/docs/developer/installation.rst

File: docs/developer/installation.rst [hunk 1]
```diff
@@ -27,37 +27,39 @@ Install spatialite and sqlite:
...
-Navigate into the cloned repository:
+The directory structure will look like this:
 
-.. code-block:: shell
+.. code-block:: text
 
-    cd openwisp-firmware-upgrader/
+    openwisp/                         # Main project folder
+    ├── env/                          # Virtual environment
+    └── openwisp-firmware-upgrader/   # Cloned repository
```
Comment (praptisharma28) [inline docs/developer/installation.rst:42]: Is not this needed then?

Comment (pandafy) [reply docs/developer/installation.rst:42]: Yes, this is not needed.

**Comment by nemesifier** [comment]
This PR has overwritten a branch which we use for storing images and GIFs shown in the docs and website.
Please do not overwrite the `docs` branch. I am closing this PR so we can restore the original content. Please open a PR against master.

**Pull request closed without merging**